src/audio/copier/copier.c

```diff
@@ -1694,13 +1700,20 @@ static int copier_get_attribute(struct comp_dev *dev, uint32_t type, void *value
 static int copier_position(struct comp_dev *dev, struct sof_ipc_stream_posn *posn)
 {
 	struct copier_data *cd = comp_get_drvdata(dev);
+	int ret = 0;
```

Comment:
init to 0 is not required now that you have ret set in both if {} and else {}

Reply:
both you and Guennadi mentioned the superfluous init, I can remove it, just have two comments here:

1. one of coding rule is init before use, I do have a issue that ret is not initialized and have no issue, but after a change inside function, ret is not always assigned, and maybe a random data in stack, and then cause a bug.

2. If not init, then it depend on developer to walk through code make sure ret is always assigned, when function is complex, something wrong may happen.

Question is:is this a rule for SOF?"if developer found no need to init, then remove the init"

Reply:
I can provide a quick and *approximate* answer, I'm sure some can offer a more comprehensive and precise one:
1. imagine all automatic variables being initialised. That would be both (subjectively) ugly and inefficient. Especially if you have larger objects on stack like arrays or large structures, that are already deterministically and provably only read after being written to.
2. yes, it is an explicit wish (I think it's also in some of coding style guides) to avoid any superfluous code. It's true that systems only ever get faster, but we're still doing real-time processing on relatively slow DSPs... This is also the style, used in the Linux kernel, where it probably also comes from
3. one of the arguments I saw for not initialising variables without need is: why initialise them to 0? Why is 0 better than ~0 or 1? An unexpected and invalid 0 can cause a bug just like any other random value. You initialise variables when it's needed and you initialise them to the values that make sense, not just to 0 because - well - we like zeros. That's some kind of numeric discrimination.
4. concerning making code (needlessly) more complex. Well, this might be a less popular opinion, but personally I'm not necessarily always in favour of making the code as simple as possible. There's more than one way to write the same program (C) and as long as that program is sufficiently complex, every programmer on the planet will come up with a different solution. I would not necessarily go with "the simplest" one of them. Probably not with "the most obscure" either, and that balance between simplicity, efficiency and elegance is highly subjective...